### DISCUSSION TITLE
Suggestion: Allow us to pass in multiple files to the Copilot Chat window

### DISCUSSION THREAD

**Discussion by Iheuzio** [community/community#69851] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Hi,

So last summer I have written an extension for [Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=Iheuzio.gpt-contextfiles), it uses GPT and OpenAI api keys to allow a user to pass in multiple files to the context of your interaction with the LLM.

I was wondering why Microsoft hasn't integrated a solution into their Copilot Chat, it will save a ton of time when working with many files and allow us to modularize our files better. This is all whilst keeping Copilot aware of our project context, without manually highlighting the text necessary to pass into the project.

If you want to use my implementation as reference, that's great. I just wish for this feature to be implemented as it'll prove greatly beneficial for the use of Copilot as a service.

Thanks,

My repository is over at: https://github.com/Iheuzio/gpt-contextfiles

**Comment by mindplay-dk** [comment]
I keep hearing how great Copilot is, and I keep trying it out, but always end up turning it off after a few days of frustration.

To be blunt, I don't really see how Copilot chat is even useful without the ability to look, at least, at a few files at a time?

Whenever I ask a question, the standard answer is "I can't answer that unless you provide more context" - the `/help` doesn't explain how to do that, and I google it every time, poke through the official docs, etc... but it doesn't look like there is currently any way to provide more context.

Correct me if I'm wrong, but it really doesn't seem like Copilot Chat is useful for anything more than trivial, stand-alone scripts at the moment? The prompts to "add more context" seem misleading, and the overall experience is just a frustrating and confusing waste of time.

They *really* need to address this.

**Reply by MelbourneDeveloper** [reply]
💯

**Reply by outgaze** [reply]
you can try doing ```@workspace``` and then asking it a question or ctrl + i for opening the inline chat but even then, they promised gpt 4 i have done some debugging with the code of the extension and forcing it to use it for code generation (if u don't do that it will be switching between gpt 4 for reasoning and 3.5 turbo for code generation) XD it's really decapitated compared to the actual gpt 4 that was in chatgpt not to mention it does not seem like they plan on using gpt 4 turbo any time sooner which is actually a bummer we are gonna be stuck with the 2021 hallucinations for a while

**Reply by realjjaveweb** [reply]
also \#file was now added, but it's still similarly unhelpful if the files are too large. (even if the task is fairly simple - like generating a XML export based on provided XML structure and description from CMS)...

**Reply by tinder-tannerbennett** [reply]
I tried using `#file` and it said `The links you provided don't contain any code.`

Edit: making a "new chat" fixed it

**Reply by sVedantWork** [reply]
if you put all the files under a single folder; you can now use "@workspace" command and mention stuff like folder_X/filey.txt etc. to make the conversation more useful.

**Reply by realjjaveweb** [reply]
@tinder-tannerbennett I've encountered numerous problems and stuck ups with the copilot chat - but all were resolved by one of these 2:
1. new chat
2. update extension/plugin and/or restart IDE

Btw nice article on @workspace : https://medium.com/@yar.dobroskok/github-copilot-workspace-new-development-experience-d69857fbd067

**Comment by rubinsh** [comment]
you can now do that using the #file comand. VSCode will ask you for the file, and you can do that with multiple files.
all these files will be passed as a reference to the prompt.
Example prompt in the co-pilot chat window:
```
#file:cashSales.test.ts #file:creditSales.test.ts can you tell me what these files do?
```

**Reply by devmi** [reply]
If you're using Copilot for VS Code, you can [Add chat context](https://code.visualstudio.com/docs/copilot/chat/copilot-chat#_add-chat-context) and select the `codebase` option or add `#codebase` in your prompt (make sure to enable the `github.copilot.chat.codesearch.enabled`)

![Add chat context](https://code.visualstudio.com/assets/docs/copilot/chat/copilot-chat/chat-add-context.png)

**Reply by devmi** [reply]
Request for this feature on IntelliJ Idea Copilot plugin:
https://github.com/orgs/community/discussions/110614#discussioncomment-13019035
